pkg/scheduler/scheduler.go

```diff
@@ -72,58 +78,116 @@ func (s *Scheduler) Less(pod1, pod2 *framework.QueuedPodInfo) bool {
 }
 
+// PreFilterExtensions returns a PreFilterExtensions interface if the plugin implements one,
+// or nil if it does not. A Pre-filter plugin can provide extensions to incrementally
+// modify its pre-processed info. The framework guarantees that the extensions
+// AddPod/RemovePod will only be called after PreFilter, possibly on a cloned
+// CycleState, and may call those functions more than once before calling
+// Filter again on a specific node.
 func (s *Scheduler) PreFilterExtensions() framework.PreFilterExtensions {
 	return nil
 }
 
+// Filter is called by the scheduling framework.
+// All FilterPlugins should return "Success" to declare that
+// the given node fits the pod. If Filter doesn't return "Success",
+// it will return "Unschedulable", "UnschedulableAndUnresolvable" or "Error".
+// For the node being evaluated, Filter plugins should look at the passed
+// nodeInfo reference for this particular node's information (e.g., pods
+// considered to be running on the node) instead of looking it up in the
+// NodeInfoSnapshot because we don't guarantee that they will be the same.
+// For example, during preemption, we may pass a copy of the original
+// nodeInfo object that has some pods removed from it to evaluate the
+// possibility of preempting them to schedule the target pod.
+func (s *Scheduler) Filter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+	return framework.NewStatus(framework.Success, "")
+}
+
+// PostFilter is called by the scheduling framework.
+// A PostFilter plugin should return one of the following statuses:
+// - Unschedulable: the plugin gets executed successfully but the pod cannot be made schedulable.
+// - Success: the plugin gets executed successfully and the pod can be made schedulable.
+// - Error: the plugin aborts due to some internal error.
+//
+// Informational plugins should be configured ahead of other ones, and always return Unschedulable status.
+// Optionally, a non-nil PostFilterResult may be returned along with a Success status. For example,
+// a preemption plugin may choose to return nominatedNodeName, so that framework can reuse that to update the
+// preemptor pod's .spec.status.nominatedNodeName field.
 func (s *Scheduler) PostFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod,
 	filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
 	return nil, nil
 }
 
+// Score is called on each filtered node. It must return success and an integer
+// indicating the rank of the node. All scoring plugins must return success or
+// the pod will be rejected.
+func (s *Scheduler) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	super, _, err := s.podGroupManager.PodGroups(pod)
+	if err != nil {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("cannot get pod group: %s", err.Error()))
```

Comment:
How about a pod with no podGroup binding, It belongs to the majority of cases.

Reply:
No, the `PodGroup` method won't return an error if you try to get a pod with no podGroup binding.

Reply:
I forget it, thanks.